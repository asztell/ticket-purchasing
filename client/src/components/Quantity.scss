.Quantity {
  .Container {
    display: flex;
    align-items: center;
    > button {
      padding: 10px 15px;
    }
  }
  .Value {
    margin-left: 20px;
    margin-right: 20px;
    font-size: 20px;
    line-height: 1;
  }
}
