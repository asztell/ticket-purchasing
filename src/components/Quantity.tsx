import { useCallback, useContext } from "react";
import { TicketPurchasingContext } from "../contexts";

export function Quantity() {
  const { ticketsCounter, updateTicketsCounter } = useContext(
    TicketPurchasingContext
  );

  const handleDecrement = useCallback(() => {
    if (ticketsCounter > 0) updateTicketsCounter(-1);
  }, [updateTicketsCounter, ticketsCounter]);

  const handleIncrement = useCallback(() => {
    updateTicketsCounter(1);
  }, [updateTicketsCounter]);

  return (
    <div className="Quantity">
      <h2>Quantity</h2>
      <div>
        <button
          id="decrement-btn"
          style={{ marginRight: "10px" }}
          onClick={handleDecrement}
        >
          -
        </button>
        <span id="counter-value">{ticketsCounter}</span>
        <button
          id="increment-btn"
          style={{ marginLeft: "10px" }}
          onClick={handleIncrement}
        >
          +
        </button>
      </div>
    </div>
  );
}
