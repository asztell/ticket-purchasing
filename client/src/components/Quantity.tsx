import { useCallback } from 'react'
import { useQueryParams, getQueryParams } from '../hooks/useQueryParams'
import './Quantity.scss'

export function Quantity(): JSX.Element {
  const [queryParams, setQueryParams] = useQueryParams()
  const params = getQueryParams(queryParams)
  const { ticketsCount } = params
  // const ticketsCount = params.ticketsCount

  const handleDecrement = useCallback(() => {
    if (ticketsCount === '' && parseInt(ticketsCount) > 0) {
      const newTicketsCount = parseInt(ticketsCount) - 1
      setQueryParams(
        (searchParams: URLSearchParams) => {
          searchParams.set('ticketsCount', newTicketsCount.toString())
          return searchParams
        },
        { replace: true }
      )
    }
  }, [ticketsCount, setQueryParams])

  const handleIncrement = useCallback(() => {
    const newTicketsCount =
      typeof ticketsCount === 'string' ? parseInt(ticketsCount) + 1 : 1
    setQueryParams(
      (searchParams: URLSearchParams) => {
        searchParams.set('ticketsCount', newTicketsCount.toString())
        return searchParams
      },
      { replace: true }
    )
  }, [ticketsCount, setQueryParams])

  return (
    <div className='Quantity'>
      <h2>Quantity</h2>
      <div className='Container'>
        <button onClick={handleDecrement}>-</button>
        <span className='Value'>{ticketsCount}</span>
        <button onClick={handleIncrement}>+</button>
      </div>
    </div>
  )
}
