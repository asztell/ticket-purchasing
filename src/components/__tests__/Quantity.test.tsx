import { render } from "@testing-library/react";
import { Quantity } from "../Quantity";

describe("<Quantity />", () => {
  test("render Quantity", () => {
    const { container } = render(<Quantity />);
    expect(container).toMatchSnapshot();
  });
});
