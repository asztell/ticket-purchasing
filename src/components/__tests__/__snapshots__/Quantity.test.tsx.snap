// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quantity /> render Quantity 1`] = `
<div>
  <div
    class="Quantity"
  >
    <h2>
      Quantity
    </h2>
    <div>
      <button
        id="decrement-btn"
        style="margin-right: 10px;"
      >
        -
      </button>
      <span
        id="counter-value"
      />
      <button
        id="increment-btn"
        style="margin-left: 10px;"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
